lib/elixir/lib/kernel.ex

```diff
@@ -2593,25 +2636,42 @@ defmodule Kernel do
         raise ArgumentError, "cannot set attribute @#{name} inside function/macro"
 
       name == :behavior ->
-        :elixir_errors.warn env.line, env.file,
-                            "@behavior attribute is not supported, please use @behaviour instead"
+        warn_message = "@behavior attribute is not supported, please use @behaviour instead"
+        :elixir_errors.warn(env.line, env.file, warn_message)
 
       # TODO: Remove :compile check once on 2.0 as we no longer
       # need to warn on parse transforms in Module.put_attribute.
       name == :compile ->
         {stack, _} = :elixir_quote.escape(env_stacktrace(env), false)
-        quote do: Module.put_attribute(__MODULE__, unquote(name), unquote(arg),
-                                       unquote(stack), unquote(line))
+
+        quote(
```

Comment:
Please convert it to a do/end block.

Reply:
I tried this, but even with do/end block it is still a couple chars do long. I kept the block for better readability though.